modules/@apostrophecms/admin-bar/ui/apos/apps/force-refresh.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * If the delivered page has a logged-out content, but a cookie say "Hey, I'm logged in",
+ * we force-refresh the page to bypass the cache, in order to get the logged-in content.
+ */
+export default () => {
+  const isLoggedInPageContent = document.body.getAttribute('data-apos-user-logged-in');
+  console.log('isLoggedInPageContent', isLoggedInPageContent);
+
+  if (isLoggedInPageContent) {
+    return;
+  }
+
+  if (apos.util.getCookie('loggedIn')) {
+    console.log('loggedIn cookie, RELOADING...');
+    apos.util.deleteCookie('loggedIn');
```

Comment:
You could encounter more than one cached page as you move through the site, so deleting the cookie is potentially problematic. But I recognize that the cookie could get out of sync with your actual login status, so we can't keep refreshing forever in a loop. So I think this is the best we can do — as long as we also *set* the cookie server-side when not already present on every newly generated response for a logged-in user. I see you haven't done that part yet.